cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/Transaction.hs

```diff
@@ -0,0 +1,110 @@
+{-# LANGUAGE DisambiguateRecordFields #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE NumericUnderscores #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+{-# OPTIONS_GHC -Wno-unrecognised-pragmas #-}
+
+{- HLINT ignore "Redundant id" -}
+{- HLINT ignore "Redundant return" -}
+{- HLINT ignore "Use head" -}
+{- HLINT ignore "Use let" -}
+
+module Cardano.Testnet.Test.Cli.Babbage.Transaction
+  ( hprop_transaction
+  ) where
+
+import           Cardano.Api
+
+import           Cardano.Testnet
+
+import           Prelude
+
+import           Control.Monad (void)
+import qualified Data.Text as Text
+import           System.FilePath ((</>))
+import qualified System.Info as SYS
+
+import           Hedgehog (Property)
+import qualified Hedgehog.Extras.Test.Base as H
+import qualified Hedgehog.Extras.Test.File as H
+
+import qualified Data.Map as Map
+import qualified Hedgehog.Extras.Test.Concurrent as H
+import           Testnet.Components.SPO
+import qualified Testnet.Process.Run as H
+import           Testnet.Process.Run
+import qualified Testnet.Property.Utils as H
+import           Testnet.Runtime
+
+hprop_transaction :: Property
+hprop_transaction = H.integrationRetryWorkspace 0 "conway-stake-snapshot" $ \tempAbsBasePath' -> do
+  H.note_ SYS.os
+  conf@Conf { tempAbsPath } <- H.noteShowM $ mkConf tempAbsBasePath'
+  let tempAbsPath' = unTmpAbsPath tempAbsPath
+  work <- H.createDirectoryIfMissing $ tempAbsPath' </> "work"
+
+  let
+    tempBaseAbsPath = makeTmpBaseAbsPath $ TmpAbsolutePath tempAbsPath'
+    era = BabbageEra
+    options = cardanoDefaultTestnetOptions
+      { cardanoNodes = cardanoDefaultTestnetNodeOptions
+      , cardanoSlotLength = 0.1
+      , cardanoNodeEra = AnyCardanoEra era -- TODO: We should only support the latest era and the upcoming era
+      }
+
+  TestnetRuntime
+    { testnetMagic
+    , poolNodes
+    , wallets
+    } <- cardanoTestnet options conf
+
+  poolNode1 <- H.headM poolNodes
+
+  poolSprocket1 <- H.noteShow $ nodeSprocket $ poolRuntime poolNode1
+
+  execConfig <- H.mkExecConfig tempBaseAbsPath poolSprocket1
+
+  let sbe = ShelleyBasedEraBabbage
```

Comment:
Defining these two eras separately could lead to issues if only one is changed. The following would be better:
```
let sbe = ShelleyBasedEraBabbage
    era = toCardanoEra sbe
```

Reply:
Fixed!